Extensions/xkit_preferences.js

```diff
@@ -950,7 +950,9 @@ XKit.extensions.xkit_preferences = new Object({
 
 			for(var extension in mdata.extensions) {
 
-				m_html = m_html + XKit.extensions.xkit_preferences.gallery_parse_item(mdata.extensions[extension]);
+				if(mdata.extensions[extension].hidden != "true"){
```

Comment:
The thing is set up to show the internal extensions in the gallery if you have the internals flag set in options.
We also had the problem before I made the PR to show all extensions that it was showing xkit_installer as installable extension